crawl-ref/source/initfile.cc

```diff
@@ -1748,7 +1751,7 @@ void save_player_name()
 void read_options(const string &s, bool runscript, bool clear_aliases)
 {
     StringLineInput st(s);
-    Options.read_options(st, runscript, clear_aliases);
+    string test = Options.read_options(st, runscript, clear_aliases);
```

Comment:
Feel free to tell me of a better way to do this, because it seems incredibly messy to create a variable which is never used again.

Reply:
I'm not sure why you create a variable here; can't you just ignore the return value?

In any case, I'm not a huge fan of teetering abstraction, but you could create a class deriving from `FileLineReader` and overriding `get_line` that additionally fills a string with the file contents. The API would look like this:
```c
CopyingFileLineReader cflr(f, Options.file_contents); // second param is the string to fill
Options.read_options(cflr, runscript);
```
If you do this, please get the file size and pre-expand the string before filling it.

Reply:
Is this necessary, because this would probably take a while for me to properly understand?

Reply:
Ok so if I understand this correctly, that would actually just be adding those two lines instead of `Options.read_options(blargh)`? I will do that then.

Reply:
Never mind, it isn't that easy; I'll leave it how it is currently, and just add the command line extra options into `Options.file_contents`.

Reply:
If you're less familiar with C++, creating such a class might be more difficult.

A third option, probably better than the current, is to read the file contents into a string, construct a StringLineInput from that, and pass that to Options.read_options instead of the FileLineReader currently used; then you have the file contents as a string as a side effect.